spring-cloud-gcp-pubsub/src/test/java/org/springframework/cloud/gcp/pubsub/support/DefaultSubscriberFactoryTests.java

```diff
@@ -70,4 +71,19 @@ public void testNewDefaultSubscriberFactory_nullProject() {
 		new DefaultSubscriberFactory(() -> null);
 	}
 
+	@Test
+	public void testCreatePullRequest_nonZeroMaxMessages() {
+		DefaultSubscriberFactory factory = new DefaultSubscriberFactory(() -> "project");
+		factory.setCredentialsProvider(this.credentialsProvider);
+
+		// If a maxMessages is null (it was omitted), then set max to MAX_INT
+		PullRequest request = factory.createPullRequest("test", null, true);
+		assertThat(request.getMaxMessages()).isEqualTo(Integer.MAX_VALUE);
+
+		// If maxMessages < 0, should throw an error.
```

Comment:
This is a nit, but can this be a different test? Null and negative numbers are distinct scenarios.

Reply:
Done.